### PR TITLE
try to find a short collision first

### PR DESCRIPTION
**Pull request by ztNFny**
massive performance improvements for larger maxlength

### CODE REVIEW

**Comment by Krevo** [comment]
Hi,

I won't accept this pull request in its current state because there is too much code duplication. 

In order to solve this problem, I've put all functions in a new file `ucommons.php.` 
Please add a `require 'ucommons.php'`in your `finduanswer_web.php`, then call `findHash()` in your code (you can have a look at the new version of `finduanswer.php` which is greatly simplified)

Please note that it is unlikely that you will find an answer with a length of 1, 2 or 3 chars, but you will very very probably find an answer with a length of 4. So your loop should start at `j=4` and `break` when the array returned by the `findHash()` function is not empty.

In short : 
- rebase on master
- rewrite `finduanswer_web.php`  with a `require 'ucommons.php'` and a call to `findHash()` 

Thanks for your interest in my tools related to Wherigo.

**Comment by ztNFny** [comment]
Actually finduanswer_web.php wasn't supposed to be included in that pull request, Github tried to be intelligent there :-( I don't think it's of any use for this project to have a web-enabled version as the decompilation relies on the unluac.jar, it's just for my personal use.

**Comment by Krevo** [comment]
Ok, totally agree with what you said about a potential web-based version.
